test/Microsoft.ML.Tests/Scenarios/Api/AspirationalExamples.cs

```diff
@@ -60,5 +60,98 @@ public void FirstExperienceWithML()
                 PetalLength = 5.1f
             });
         }
+
+
+        [Fact]
+        public void SimpleIrisDescisionTrees()
+        {
+            var env = new TlcEnvironment(new SysRandom(0), verbose: true);
+            string dataPath = "iris-data.txt";
+            // Create reader with specific schema.
+            var dataReader = TextLoader.CreateReader(env, ctx => (
+               Label: ctx.LoadText(0),
+               SepalWidth: ctx.LoadFloat(1),
+               SepalLength: ctx.LoadFloat(2),
+               PetalWidth: ctx.LoadFloat(3),
+               PetalLength: ctx.LoadFloat(4)),
+               dataPath);
+
+            // Load the data into the system.
+            var data = dataReader.Read(dataPath);
+
+            var preprocess = data.Schema.MakeEstimator(row => (
+                // Convert string label to key.
+                Label: row.Label.Dictionarize(),
+                // Concatenate all features into a vector.
+                Features: row.SepalWidth.ConcatWith(row.SepalLength, row.PetalWidth, row.PetalLength)));
+
+
+            var pipeline = preprocess
+                .Append(row => row.Label.PredictWithDecisionTrees(row.Features))
+                // shoul it be BagVectorize instead of KeyToValue?
+                .Append(row => row.PredictedLabel.KeyToValue());
+
+            var model = pipeline.Fit(data);
+
+            var predictions = model.Transform(dataReader.Read(testDataPath));
+            var evaluator = new MultiClassEvaluator(env);
+            var metrics = evaluator.Evaluate(predictions);
+        }
+
+        [Fact]
+        public void TwitterSentimentAnalysis()
+        {
+            var env = new TlcEnvironment(new SysRandom(0), verbose: true);
+            var dataPath = "wikipedia-detox-250-line-data.tsv";
+            // Load the data into the system.
+            var data = TextLoader.CreateReader(env, ctx => (
+                   Label: ctx.LoadFloat(0),
+                   Text: ctx.LoadText(1)),
+                   dataPath, hasHeader: true).Read(dataPath);
```

Comment:
This is something that is slightly annoying me as I'm reading across the examples, and I wonder if anyone else feels similarly.

This `.CreateReader(..., dataPath).Read(dataPath)` seems vaguely tautological -- it probably isn't, but it seems like it is. While surely we must have `DataReader` be able to handle the case of different data, would there be benefit to having a `DataReader` somehow retain a reference to the `TIn` instance that it was initialized on, and so have a parameterless `Read` that just implicitly read over whatever it was initialized as?

Nonetheless, you could imagine this being something we would not want to do all the time -- if the initialization structure was some huge in-memory structure (e.g., the case where we are initializing and training a pipeline based on data in memory). But then we have to have a separate API to allow both cases. And if we don't want to do that *always*, perhaps it's best if we don't do it *ever*, so a user does not have to learn additional special cased idioms for what amounts to a fairly minor enhancement.

With that in mind, I think the code in its current form is probably fine (since there are some strong architectural reasons to disfavor caching, as we see above), but at the very least we will need to make sure this "idea" comes across somehow perhaps better than it does now, since at first glance this looks, frankly, kind of silly. I wonder at least if there's something we can do to make it seem, somehow, less silly.